event-svc/src/event/validator/event.rs

```diff
@@ -0,0 +1,151 @@
+use std::sync::Arc;
+
+use ceramic_core::{Cid, EventId};
+use ceramic_event::unvalidated;
+use ipld_core::ipld::Ipld;
+use recon::ReconItem;
+
+use crate::{
+    event::{service::InvalidItem, DeliverableRequirement},
+    store::{EventInsertable, SqlitePool},
+    Result,
+};
+
+#[derive(Debug)]
+pub struct ValidatedEvents {
+    /// These events are valid
+    pub valid: Vec<ValidatedEvent>,
+    /// We don't have enough information to validate these events yet.
+    /// e.g. The init event is required and the node has not yet discovered it.
+    pub pending: Vec<UnvalidatedEvent>,
+    /// Events that failed validation
+    pub invalid: Vec<InvalidItem>,
+}
+
+#[derive(Debug)]
+pub struct ValidatedEvent {
+    key: EventId,
+    cid: Cid,
+    event: Arc<unvalidated::Event<Ipld>>,
+}
+
+impl ValidatedEvent {
+    pub fn order_key(&self) -> &EventId {
+        &self.key
+    }
+}
+
+// TODO: should this impl from or have a private method?
+// it's not supposed to be public outside the crate and it currently allows us to
+// skip validation for the cases we're not doing it
+impl From<UnvalidatedEvent> for ValidatedEvent {
+    fn from(value: UnvalidatedEvent) -> Self {
+        Self {
+            key: value.key,
+            cid: value.cid,
+            event: value.event,
+        }
+    }
+}
+
+impl From<ValidatedEvent> for EventInsertable {
+    fn from(value: ValidatedEvent) -> Self {
+        EventInsertable::new(value.key, value.cid, value.event, false)
+            .expect("validated events must be insertable")
+    }
+}
+
+#[derive(Debug)]
+pub struct UnvalidatedEvent {
```

Comment:
yeah, I could go either way (either naming the pending stuff in service "unvalidated" or renaming this). it may turn into either a "pending event" or an "invalid event" if the current context requires local history (e.g. an api write).

Reply:
Same with this one, I think unvalidated makes the most sense and Pending is context dependent. I think it's a bit more clear than it was, but it can be better so I can look at it in #503 or another one.